test/e2e/create_cluster_test.go

```diff
@@ -1396,6 +1421,98 @@ func TestCreateClusterPrivateWithRouteKAS(t *testing.T) {
 	testCreateClusterPrivate(t, true)
 }
 
+// TestCreateClusterWithDisabledCapabilities implements a test that creates a cluster
+// with the ImageRegistry capability disabled, then attempts to enable it.
+func TestCreateClusterWithDisabledCapabilities(t *testing.T) {
```

Comment:
we'll want to avoid increasing the cloud load by creating yet another hc for this. Instead we can reuse the existing TestCreateClusterCustomConfig to include these validations

Reply:
We talked about this on Slack: I'll drop the e2e from this PR and implement this comment on a separate PR, where we'll also drop the tech preview feature gate.